SU-HSDO/suhumsci
# SHS-5059: Skip to main content

# READY FOR REVIEW

## Summary
- Update `page.html.twig` templates to include the `main-content` id only when layout builder is not enabled
- Add field a logic to include the `main-content` id on the layout builder sections. The id can be added on the main region, around the whole section to include the left sidebar or disabled
- Add validation to the layout customizing forms (both default and overridden) to ensure that the `main-content` id is present in exactly one section
- Create update hooks to update the sections main content field for both default and overridden layouts. For all layouts, the first non full width layout is being configured to display the `main-content` id in the main region. In all other sections in the layout, the main content field is set to `none`.
- Use `<main>` tags for the main content container.

## Need Review By (Date)
11/13

## Urgency
medium

## Steps to Test
1. After the site is installed, visit the default layout settings (Manage display > Manage Layout) for all content types and confirm that the "Main Content begins at" field is present in all section variants. Confirm that, for all content types, the first non full width section has that field set to "Main region" and all others sections has it set to "None".
2. Find a flexible page or private page with an overridden layout. Confirm that the new field is present and that the configurations is the same as the default layouts (first non full width section set to "Main region", all others to "None"). Some examples of nodes with overridden layout in traditional:
  a. https://hs-traditional.suhumsci.loc/mixed-page-well-raised-cards-full-width
  b. https://hs-traditional.suhumsci.loc/mixed-pages/mixed-page-wells-raised-cards-full-width-sidebar
  c. https://hs-traditional.suhumsci.loc/components/collections-cards
  d. https://hs-traditional.suhumsci.loc/private-pages/private-page-collection-cards
3. Edit a default layout and try the following:
  a. Set the main content field in all sections to "None"
  b. Set more than one section with a value different to "None"

    In both cases, you should get a validation error that prevents you to save the form until exactly one section has a non "None" value.

4. Repeat the previous tests on a node with an overridden layouts. You should get the same validation.
5. Test some random pages on the site. All of them should have exactly one `main-content` id on its source, in the position configured on the layout settings for that content type or node. The container with the `main-content` id should use the `<main>` tag.

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Humsci Basic PR Checklist](https://github.com/SU-HSDO/suhumsci/blob/develop/docs/HumsciBasicPRChecklist.md)
